src/Sylius/Component/Locale/Provider/CachedLocaleCollectionProvider.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Locale\Provider;
+
+use Symfony\Contracts\Cache\CacheInterface;
+
+final class CachedLocaleCollectionProvider implements LocaleCollectionProviderInterface
+{
+    public function __construct(private LocaleCollectionProviderInterface $decorated, private CacheInterface $cache)
+    {
+    }
+
+    public function getAll(): array
+    {
+        return $this->cache->get('sylius_locales', function () {
+            return $this->decorated->getAll();
+        });
```

Comment:
We've talked about it but I'll note it so that it won't disappear;
There needs to be some cache-busting mechanism in place, otherwise, we'll be potentially stuck with the same locales' codes forever, regardless of whether an admin adds more.